python-package/lightgbm/engine.py

```diff
@@ -125,6 +127,20 @@ def train(
     """
     # create predictor first
     params = copy.deepcopy(params)
+    eval_metric = params.get('metric')
```

Comment:
There are some other names (aliases) for the `metric` param. We should check all of them
https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric

Also, the same is true for the `objective` param.
https://lightgbm.readthedocs.io/en/latest/Parameters.html#objective

Reply:
got it